src/treelearner/gpu_tree_learner.cpp

```diff
@@ -52,7 +52,7 @@ void PrintHistograms(hist_t* h, size_t size) {
   double total_hess = 0;
   for (size_t i = 0; i < size; ++i) {
     printf("%03lu=%9.3g,%9.3g\t", i, GET_GRAD(h, i), GET_HESS(h, i));
-    if ((i & 2) == 2)
+    if ((i & 3) == 3)
```

Comment:
there are only grad and hess now, why change it to `3`?

Reply:
This seems like a logic bug.  The original code prints TWO LINE FEEDS for every 4 values (2, 3, 6, 7, 10, 11, etc).  It seems like the intent of this code is print ONE LINE FEED every 4 values (3, 7, 11, etc).

Reply:
Chip is correct.  I made this change because if you "play computer" on the red (deleted) line, you get a funny printout as he described.  This is about ensuring that we have one line feed for every four values (print four per line).

The green line is the moral equivalent of "if ((i % 4) == 3)"